library/src/main/java/com/voysis/Extensions.kt

```
@@ -0,0 +1,43 @@
+package com.voysis
+
+import android.media.AudioFormat
+import android.media.AudioRecord
+import android.os.Build
+import com.voysis.recorder.MimeType
+
+fun AudioRecord.generateMimeType(): MimeType {
+    return MimeType(sampleRate = sampleRate,
+            bitsPerSample = getBitsPerSample(),
+            bigEndian = false,
+            channels = channelCount,
+            encoding = getEncodingString())
+}
+
+fun AudioRecord.getBitsPerSample(): Int {
+    if (Build.VERSION.SDK_INT >= 21) {
+        return when (audioFormat) {
+            AudioFormat.ENCODING_PCM_FLOAT -> 32
+            AudioFormat.ENCODING_PCM_16BIT -> 16
+            AudioFormat.ENCODING_PCM_8BIT -> 8
+            else -> {
+                -1
```

I think you should straight up throw an exception here an break, rather than passing an invalid number of bits per sample to the MIME type.